### PR TITLE
Escape latex and html special chars in repr functions

### PR DESCRIPTION
**Pull request by janschulz**
Bad things happen if a character vector has valid html included and the valid
html is not escaped. Same for latex...

### CODE REVIEW

File: R/repr_vector.r [hunk 1]
```diff
@@ -146,7 +151,8 @@ repr_latex.logical <- function(obj, ...) repr_vector_generic(
 	'\\textbf{%s:} %s',
```
Comment (janschulz) [inline R/repr_vector.r]: I'm not sure if the `latex.escape.names(obj)` could also be changed into `obj`:
```R
repr_latex.logical <- function(obj, ...) repr_vector_generic(
 	latex.escape.names(obj),  # escape vector names, regardless of class
```

**Comment by flying-sheep** [comment]
cool. can you add more test?

that some kinds of HTML & LaTeX are escaped in rownames, colnames, list names, vector names, vector values, list values and matrix values?

**Comment by takluyver** [comment]
Thanks for tackling this :smiley:

**Comment by janschulz** [comment]
Lets see... My latex is rusty, I've actually no idea if the produces latex code is correct or not :-(

**Comment by flying-sheep** [comment]
well, you can assume that [this list](https://github.com/IRkernel/repr/blob/da0ce5977621c574d054e58e8fece3df58102af8/R/utils.r#L10) is correct and complete.*

so characters on the left should be replaced with the strings on the right when going from raw data → LaTeX

---

*this is in fact wrong, as you can always change catcodes in LaTeX, but all packages assume that people don’t mess with catcodes so we can as well.

**Comment by janschulz** [comment]
The problem is more if I would catch if the code would double escape stuff :-)

**Comment by flying-sheep** [comment]
you do, but i fixed it

**Comment by flying-sheep** [comment]
merged. dunno why GH didn’t pick it up

**Pull request closed without merging**